python/setup.py

```diff
@@ -104,13 +96,14 @@ def run(self):
     # github.com/libdynd/dynd-python
 
     description = "Build the C-extensions for arrow"
-    user_options = ([('extra-cmake-args=', None,
-                      'extra arguments for CMake')] +
-                    _build_ext.user_options)
+    user_options = ([('extra-cmake-args=', None, 'extra arguments for CMake'),
+                     ('build-type=', None, 'build type (debug or release)')]
+                    + _build_ext.user_options)
 
     def initialize_options(self):
         _build_ext.initialize_options(self)
         self.extra_cmake_args = os.environ.get('PYARROW_CMAKE_OPTIONS', '')
+        self.build_type = 'debug'
```

Comment:
Could you do me a favor and pull here from an environment variable, i.e. `os.environ.get('PYARROW_BUILD_TYPE', 'debug')`?